proof/meta.go

```diff
@@ -306,59 +305,39 @@ func (m *MetaReveal) DecDisplayOption() (fn.Option[uint32], error) {
 
 // SetDecDisplay attempts to set the decimal display value in existing JSON
 // metadata. It checks that the new metadata is below the maximum metadata size.
-func (m *MetaReveal) SetDecDisplay(decDisplay uint32) (*MetaReveal, error) {
```

Comment:
IIUC this was in place to not mutate the meta reveal receiver here. AFAICT, was mainly a defensive precaution.